src/win/tty.c

```diff
@@ -2279,3 +2275,52 @@ static void uv__determine_vterm_state(HANDLE handle) {
 
   uv__vterm_state = UV_SUPPORTED;
 }
+
+static DWORD WINAPI uv__tty_console_resize_message_loop_thread(void* param) {
+  CONSOLE_SCREEN_BUFFER_INFO screen_buffer_info;
+  MSG msg;
+
+  if (!GetConsoleScreenBufferInfo(uv_tty_console_handle, &screen_buffer_info))
+    uv_fatal_error(GetLastError(), "GetConsoleScreenBufferInfo");
+
+  uv_tty_console_width = screen_buffer_info.dwSize.X;
+  uv_tty_console_height = screen_buffer_info.srWindow.Bottom - screen_buffer_info.srWindow.Top + 1;
+
+  if (!SetWinEventHook(EVENT_CONSOLE_LAYOUT,
+                       EVENT_CONSOLE_LAYOUT,
+                       NULL,
+                       uv__tty_console_resize_event,
+                       0,
+                       0,
+                       WINEVENT_OUTOFCONTEXT))
+    uv_fatal_error(GetLastError(), "SetWinEventHook");
+
+  while (GetMessage(&msg, NULL, 0, 0)) {
```

Comment:
I've described it in https://github.com/libuv/libuv/pull/1408#issuecomment-320648292